catroid/src/main/java/org/catrobat/catroid/cloudmessaging/CloudMessaging.java

```diff
@@ -54,6 +69,50 @@ public void showNotification(CloudMessage cloudMessage) {
 		}
 	}
 
+	public static boolean initialize(Context context) {
+		if (isCloudNotificationsEnabled(context)) {
+			subscribeToFirebaseServices(context);
+			return true;
+		}
+		unsubscribeFromFirebaseServices(context);
+		return false;
+	}
+
+	@VisibleForTesting
+	public static void setIsCloudNotificationsEnabled(boolean isEnabled) {
+		isCloudNotificationsEnabled = isEnabled;
+	}
+
+	private static void subscribeToFirebaseServices(Context context) {
+		FirebaseMessaging.getInstance().subscribeToTopic(DEFAULT_TOPIC);
+		FirebaseMessaging.getInstance().subscribeToTopic(getLanguageCode(context));
+		FirebaseMessaging.getInstance().subscribeToTopic(getCountryCode(context));
+	}
+
+	private static void unsubscribeFromFirebaseServices(Context context) {
+		FirebaseMessaging.getInstance().unsubscribeFromTopic(DEFAULT_TOPIC);
+		FirebaseMessaging.getInstance().unsubscribeFromTopic(getLanguageCode(context));
+		FirebaseMessaging.getInstance().unsubscribeFromTopic(getCountryCode(context));
+	}
+
+	private static boolean isCloudNotificationsEnabled(Context context) {
+		SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context);
+		return sharedPreferences.getBoolean(SettingsActivity.SETTINGS_CLOUD_NOTIFICATIONS, false) && isCloudNotificationsEnabled;
+	}
+
+	private static String getLanguageCode(Context context) {
+		String languageCode = context.getResources().getConfiguration().locale.getCountry();
```

Comment:
> The problems, it can't be combined with topics...

I think we can't use it then. But on the other hand

> https://stackoverflow.com/a/8691091/2774399 looks interesting.

We can use it. So here we will create a key-value pair between the code and the language. As far as I have understood the concept we will be using it like this (en = "English", fr = "French") and we will be subscribing the users to values (English, French). I think key-value pairs will be created manually in strings.xml.

Another problem is subscribing/unsubscribing during the language change. We can solve in 2 ways:-
1. We can store the current locale on sharedPreferences and whenever the user opens the device we can check if stored locale is equal to the current locale.
 OR
2. https://stackoverflow.com/a/43584678/4590536 [we can use locale change listener]

Now, 
- Do I have to create key-value pairs for strings.xml?
- Which of the 2 methods (or any other method) should I use? (for un/subscribing)